units/event_log_tests.cc

```diff
@@ -205,6 +206,104 @@ std::string EventToCSVString(Event &event) {
   return csv_str.str();
 }
 
+// EventLogger base class
+//
+// EventLogger logs an event in a desired format (CSV, Chrome Trace Event, etc).
+// It receives a stream of `Event`s as input, converts them to string, and
+// passes them to the LogOutput instance which is resposible for writing the
+// logs. All loggers must implement this class based on their output format
+// specifications.
+class EventLogger {
+ public:
+  virtual ~EventLogger() = default;
+  // Converts an `Event` into a string and writes it to the output.
+  virtual void InsertEvent(Event *event) = 0;
+  // Writes the output format's prefix to the output. For instance, in case of a
+  // CSV log, the the prefix is the CSV header.
+  virtual void StartLog() = 0;
+  // Writes the output format's suffix to the output.
+  virtual void EndLog() = 0;
+  // Makes sure all the logs are written to the output.
+  virtual void Flush() = 0;
+};
+
+// FilterLogger is a wrapper around the `EventLogger` which filters the input
+// events based on a given log level.
+// Example:
+// ``` C++
+// TestLogger test;
+// FilterLogger(&test, LogLevel::High);
+// ```
+class FilterLogger : public EventLogger {
+ public:
+  FilterLogger(EventLogger *logger, LogLevel log_level)
+      : logger_(logger), log_level_(log_level){};
+
+  void InsertEvent(Event *event) override {
+    if (event->GetLogLevel() >= log_level_) logger_->InsertEvent(event);
+  }
+
+  void StartLog() override { logger_->StartLog(); }
+
+  void EndLog() override { logger_->EndLog(); }
+
+  void Flush() override { logger_->Flush(); }
+
+ private:
+  EventLogger *logger_;
+  LogLevel log_level_;
+};
+
+// A subclass of `EventLogger` that contains more than one logger and invokes
+// the same method on all of them. Example: `BroadCastLogger({&event_logger1,
+// &event_logger2});`
+class BroadcastLogger : public EventLogger {
+ public:
+  BroadcastLogger(std::vector<EventLogger *> loggers) : loggers_(loggers) {}
+  void InsertEvent(Event *event) override {
+    for (EventLogger *logger : loggers_) logger->InsertEvent(event);
+  }
+  void StartLog() override {
+    for (EventLogger *logger : loggers_) logger->StartLog();
+  }
+  void EndLog() override {
+    for (EventLogger *logger : loggers_) logger->EndLog();
+  }
+  void Flush() override {
+    for (EventLogger *logger : loggers_) logger->Flush();
+  }
+
+  std::vector<EventLogger *> &GetLoggers() { return loggers_; }
```

Comment:
Do we ever need the non-const version of this function? Can we add `const` like with `GetEvents` in `TestLogger`?